build/azure-pipelines/sql-product-build.yml

```diff
@@ -129,15 +129,30 @@ stages:
     #   timeoutInMinutes: 90
 
   - stage: Release
-    condition: and(succeeded(), or(eq(variables['VSCODE_RELEASE'], 'true'), and(eq(variables['VSCODE_QUALITY'], 'insider'), eq(variables['Build.Reason'], 'Schedule'))))
+    # Requirements:
+    # 1. Release can be created only from builds of the official build pipeline, but not all builds of it should be released automatically.
+    # 2. A build should be released only when the release variable is set to true.
+    # 3. The daily scheduled Insiders build should be released automatically.
+    #
+    # Limitation:
```

Comment:
TBH I'd just remove this limitation section. It's not super relevant anymore with the latest changes.

Reply:
I'd leave it here so that other folks who might work on this job knows why we are not using the easiest approach: use the VSCODE_RELEASE to do what it is supposed to do.

Reply:
Can you at least update the "Unfortunately ADO doesn't..." part to be 

`Unfortunately ADO doesn't allow overriding variable values for scheduled runs. (see https://learn.microsoft.com/en-us/azure/devops/pipelines/process/scheduled-triggers?view=azure-devops&tabs=yaml#scheduled-triggers). This means we can't simply set VSCODE_RELEASE to true only for scheduled builds and have to set it to true for all runs of the pipeline by default`. 

This is more accurate. 

Although thinking about it - there's no reason we couldn't have a task that checks if this is a scheduled build and if so update the VSCODE_RELEASE to true in the pipeline itself....that should actually let us keep the default as false and simplify this expression too.

Reply:
done. I'll create an issue to track the task of simplifying this.

Reply:
I will actually give it a try.

Reply:
there is no way you can achieve only release scheduled insider build from a particular pipeline, that will go back to the original problem we had today. there is a predefined variable - Build.DefinitionName, the build pipeline's name, but I don't want to hard the pipeline name into it. going to commit as is.